update defect dojo article and pass vale tool (#1233)
returntocorp/semgrep-docs: docs/kb/integrations/defect-dojo-integration.md

1. In DefectDojo:
    1. Create your [**product**](https://defectdojo.github.io/django-DefectDojo/usage/models/#products).
    2. In that DefectDojo product, create an engagement (it is a description of the import you will run), called `semgrep`. This is a CI/CD engagement type.
2. Run a semgrep scan with flags `--json --output report.json` to generate a JSON report.

> Now, you are ready to use the [DefectDojo API](https://defectdojo.github.io/django-DefectDojo/integrations/api-v2-docs/).

Author: I checked that the endpoints for the new API version: https://demo.defectdojo.org/api/v2/oa3/swagger-ui/
are just the same for the example:
'/api/v2/import-scan/'
'/api/v2/reimport-scan/'

* `PRODUCT` = chess-game
* `IMPORT_SEMGREP_TO_DEFECTDOJO_SHA_CHECKSUM` = c41aed4055adeee415b795cc17a069b144fb51bc31f6c4925be3b82d0b54de33 Uimport_semgrep_to_defect_dojo.py

The content for this last variable was generated with the following command:
`shasum -a 256 -U import_semgrep_to_defect_dojo.py`

> it will generate an unique key taking as input the content of the script and it will be used to verify the integrity. 

Reviewer: ```suggestion
This command generates a unique hash, taking as input the content of the script, and it will be used to verify that the script has not changed. 
```

In the pipeline, the integrity of the script is verified with the following commands:
```
echo $IMPORT_SEMGREP_TO_DEFECTDOJO_SHA_CHECKSUM > sha-import-dd.tmp
shasum -a 256 -U -c sha-import-dd.tmp
```

> If the script has not been tampered, the pipeline will continue normal execution, otherwise it will stop and return an error.

Reviewer: ```suggestion
If the script has not changed since the checksum was generated, the pipeline will continue normal execution, otherwise it will stop and return an error.
```
